enhancements/ingress/admin-configured-ca-bundle.md

```diff
@@ -0,0 +1,330 @@
+
+Change OpenShift Ingress Operator to:
+1. Make the CA certificate bundle defined in `IngressController` available to the OpenShift Router as the 
+   default destination CA bundle file to be used for verifying the services' certificate.
+
+Update router to:
+1. Watch the destination CA certificate bundle and update its content internally when needed.
+
+### Workflow Description
+
+- An OpenShift cluster administrator configures the list of trusted CA certificates as a ConfigMap, and
+  provides reference to it in the `IngressController` object.
+- The Ingress Operator configures OpenShift router with a ConfigMap volume mount to use the ConfigMap as 
+  the default destination CA certificate to verify a service's certificate when a route with re-encrypt
+  termination type is created with an empty destination CA.
+- OpenShift router watches the volume mount and reloads HAProxy whenever the CA certificate bundle changes.
+
+#### Variation [optional]
+None
+
+### API Extensions
+
+The proposal expects the OpenShift cluster administrator to create a ConfigMap with all the required 
+CA certificates, including the OpenShift service CA in the same namespace of the `IngressController` object
+in which the configmap reference is provided.
+
+Enhancement requires below modifications to the `IngressController` API.
+- Add `defaultDestinationCATrust` field to `.spec` of the `ingresscontrollers.operator.openshift.io`
+```golang
+type IngressControllerSpec struct {
+	// defaultDestinationCATrust contains reference to an object containing the trusted
+	// Certificate Authorities (CAs) used by the Router for verifying the services'
+	// certificates of re-encrypt routes. When Routes don't specify their own
+	// destinationCACertificate, defaultDestinationCATrust is used.
+	//
+	// The user is responsible for including OpenShift service CA along with their own
+	// trusted CA certificates in the referenced object. OpenShift service CA can be
+	// found in openshift-service-ca.crt configmap in openshift-config namespace.
+	//
+	// If unset, OpenShift service CA configured in openshift-service-ca.crt configmap will
+	// be used as default destination CA certificate.
+	//
+	// +optional
+	DefaultDestinationCATrust DefaultDestinationTrust `json:"defaultDestinationCATrust,omitempty"`
+}
+
+// DefaultDestinationTrust specifies the reference to an object containing the trusted CA certificates.
+// +kubebuilder:validation:XValidation:rule="has(self.type) && self.type == 'ConfigMap' ?  has(self.configMap) : !has(self.configMap)",message="configMap must be configured when type is ConfigMap"
+// +union
+type DefaultDestinationTrust struct {
+	// type is the referenced object type which contains the trusted CA certificates.
+	// The default value is ConfigMap. ConfigMap is the only supported type at present.
+	//
+	// +unionDiscriminator
+	// +kubebuilder:default:="ConfigMap"
```

Comment:
I suspect you'll need "" without defaulting to handle the, "I'm using the default case for NE today"